### ISSUE TITLE
flamegraph display

### ISSUE THREAD

**Issue by ColdV** [pyroscope-io/pyroscope#2936]
when i  slelect  one process, but the flamegraph still displaying other process. what should i do?
![POPO-screenshot-20240117-230418](https://github.com/grafana/pyroscope/assets/34326084/769eae85-339f-4e8f-81e9-36e00e6bd196)

**Comment by korniltsev** [comment]
share your configuration

it is probably because you are using "targets_only=false", if that is the case you should migrate to `discovery.process` and `targets_only=true`

**Comment by ColdV** [comment]
> share your configuration
> 
> it is probably because you are using "targets_only=false", if that is the case you should migrate to `discovery.process` and `targets_only=true`

i’m try,  but it didn't solve the problem. this is my grafana-agent-flow config:
![POPO-screenshot-20240117-232448](https://github.com/grafana/pyroscope/assets/34326084/f0670cff-b60f-47b9-864f-0f0937185c71)
and i not found doc about discovery.process, please send me?
and i have another question, when i only start pyroscope service by default config(no start grafana-agent-flow).  i can still see the 
pyroscope data on flamegraph. that is default display? how shoule i close it?
![POPO-screenshot-20240117-233002](https://github.com/grafana/pyroscope/assets/34326084/19e8dca4-e160-431a-8e05-092881a17e97)

**Comment by cyriltovena** [comment]
Yeah right now this only affect the Graph part of the panel. You need to use the query field `{service_name="test"}` and submit a new query.
